concordium-sdk/src/main/java/com/concordium/sdk/crypto/NativeResolver.java

```diff
@@ -19,28 +19,36 @@ public class NativeResolver {
     public static void loadLib() {
         if (!LOADED) {
             try {
-                val os = NativeResolver.OS.from(System.getProperty("os.name"));
-                val libName = os.getPrefix() + BASE_LIB_NAME + os.getExtension();
-                val libPath = "/native/" + libName;
+                if (NativeResolver.isAndroid()) {
+                    System.loadLibrary(BASE_LIB_NAME);
+                } else {
+                    val os = NativeResolver.OS.from(System.getProperty("os.name"));
+                    val libName = os.getPrefix() + BASE_LIB_NAME + os.getExtension();
+                    val libPath = "/native/" + libName;
 
-                val resourceAsStream = NativeResolver.class.getResourceAsStream(libPath);
-                if (Objects.isNull(resourceAsStream)) {
-                    throw new RuntimeException("FAILED LOADING LIB");
-                }
-                val tempLib = File.createTempFile(getRandomPrefix(), libName + os.getExtension());
-                tempLib.deleteOnExit();
+                    val resourceAsStream = NativeResolver.class.getResourceAsStream(libPath);
+                    if (Objects.isNull(resourceAsStream)) {
+                        throw new RuntimeException("FAILED LOADING LIB");
+                    }
+                    val tempLib = File.createTempFile(getRandomPrefix(), libName + os.getExtension());
+                    tempLib.deleteOnExit();
 
-                try (FileOutputStream fos = new FileOutputStream(tempLib)) {
-                    IOUtils.copy(resourceAsStream, fos);
+                    try (FileOutputStream fos = new FileOutputStream(tempLib)) {
+                        IOUtils.copy(resourceAsStream, fos);
+                    }
+                    System.load(tempLib.getAbsolutePath());
                 }
-                System.load(tempLib.getAbsolutePath());
                 LOADED = true;
             } catch (Exception e) {
                 throw new IllegalStateException("Could not load native dependencies", e);
             }
         }
     }
 
+    private static boolean isAndroid() {
+        return System.getProperty("java.runtime.name").equals("Android Runtime");
```

Comment:
Is it documented anywhere that this is the correct way of checking if code is running on Android? I couldn't find anything about it.

An alternative would be to compile a different `NativeResolver` for the Android package vs. the regular one. If the above is not well-documented or guaranteed to be stable.

Reply:
I got it from here :upside_down_face: 
https://stackoverflow.com/questions/4519556/how-to-determine-if-my-app-is-running-on-android
Might not be reliable, but it doesn't seem like there is any "official" way to do it.

I can look into the "compile-time" fix. :thinking: